tests/mr/device/mr_test.hpp

```diff
@@ -246,7 +250,13 @@ struct mr_allocation_test : public mr_test {
 /// MR factory functions
 inline auto make_cuda() { return std::make_shared<rmm::mr::cuda_memory_resource>(); }
 
-inline auto make_cuda_async() { return std::make_shared<rmm::mr::cuda_async_memory_resource>(); }
+inline auto make_cuda_async()
+{
+  if (rmm::mr::cuda_async_memory_resource::is_supported()) {
+    return std::make_shared<rmm::mr::cuda_async_memory_resource>();
+  }
+  return std::shared_ptr<rmm::mr::cuda_async_memory_resource>{};
```

Comment:
This is a bit subtle since the only difference is `()` vs. `{}`. Maybe use an explicit `nullptr`?

Reply:
Done.